streaming/base/dataset.py

```diff
@@ -127,9 +127,12 @@ class StreamingDataset(IterableDataset):
             initial run.
         batch_size (int, optional): Batch size of its DataLoader, which affects how the dataset is
             partitioned over the workers. Defaults to ``None``.
+        partition_algo (str): Which partitioning algorithm to use. Defaults to ``orig``.
+        shuffle_algo (str): Which shuffling algorithm to use. Defaults to ``py2s``.
```

Comment:
What is `py2s` here means ?

Reply:
string giving the name of a shuffling algorithm

`py2s` py for python, s for the shuffling unit is the shard, 2 for the shuffling on the basis of shards happens two times within the algorithm

the new algorithm is `py1s`

the c++ extension rewrite of that is `cc1s`

in the future, we should investigate getting better convergence for slight loss of performance by shuffling in given fixed units of samples, which would be a `py1u` etc

Reply:
Please also add the above explanation in the code for the users to know in your follow up PR. Thanks!